lib/zhong.rb

```diff
@@ -41,8 +41,19 @@ def self.latest_heartbeat
   end
 
   def self.all_heartbeats
-    Zhong::Util.safe_mget(Zhong.redis.scan_each(match: "zhong:heartbeat:*").to_a).map do |k, v|
-      host, pid = k.split("zhong:heartbeat:", 2)[1].split("#", 2)
+    heartbeat_key = scheduler.config[:heartbeat_key]
+    heartbeats = Zhong.redis.hgetall(heartbeat_key)
+    heartbeats.select do |k, v|
+      if Time.at(v.to_i) > 15.minutes.ago
+        Zhong.redis.hdel(heartbeat_key, k)
```

Comment:
I believe you can `hdel` multiple keys at once: http://redis.io/commands/hdel, so I think a slight improvement would be to collect all expired keys, and do one `hdel` at the end, and return the unexpired ones.